app/api/utility.py

```diff
@@ -24,21 +57,33 @@ def parse_nodes_as_dict(path: Path) -> dict:
     where the keys are the node URLs, and the values are the node names.
     Makes sure node URLs end with a slash.
     """
-    # TODO: Add more validation of input JSON, including for JSONDecodeError (invalid JSON)
     if path.exists() and path.stat().st_size > 0:
-        with open(path, "r") as f:
-            local_nodes = json.load(f)
-        if local_nodes:
-            if isinstance(local_nodes, list):
+        try:
+            with open(path, "r") as f:
+                local_nodes = json.load(f)
+        except json.JSONDecodeError:
+            warnings.warn(f"You provided an invalid JSON file at {path}.")
+            local_nodes = []
+
+        input_nodes = local_nodes if isinstance(local_nodes, list) else [local_nodes]
+        valid_nodes = []
+        invalid_nodes = []
+        for node in input_nodes:
+            try:
+                validate(instance=node, schema=LOCAL_NODE_SCHEMA["definitions"]["node"])
+                valid_nodes.append(node)
+            except jsonschema.ValidationError:
+                invalid_nodes.append(node)
+
+        if invalid_nodes:
+            warnings.warn(
+                f"Some of the nodes in the JSON are invalid: {invalid_nodes}"
```

Comment:
🍒: since `invalid_nodes` will be a list of dicts, is there a way to format this string so that the dicts appears properly indented? This might be a bit more readable for the user.